embassy-stm32/src/pwm/mod.rs

```diff
@@ -118,38 +123,7 @@ macro_rules! impl_compare_capable_16bit {
 
 foreach_interrupt! {
     ($inst:ident, timer, TIM_GP16, UP, $irq:ident) => {
-        impl crate::pwm::sealed::CaptureCompare16bitInstance for crate::peripherals::$inst {
-            unsafe fn set_output_compare_mode(
-                &mut self,
-                channel: crate::pwm::Channel,
-                mode: OutputCompareMode,
-            ) {
-                use crate::timer::sealed::GeneralPurpose16bitInstance;
-                let r = Self::regs_gp16();
-                let raw_channel: usize = channel.raw();
-                r.ccmr_output(raw_channel / 2)
-                    .modify(|w| w.set_ocm(raw_channel % 2, mode.into()));
-            }
-
-            unsafe fn enable_channel(&mut self, channel: Channel, enable: bool) {
-                use crate::timer::sealed::GeneralPurpose16bitInstance;
-                Self::regs_gp16()
-                    .ccer()
-                    .modify(|w| w.set_cce(channel.raw(), enable));
-            }
-
-            unsafe fn set_compare_value(&mut self, channel: Channel, value: u16) {
-                use crate::timer::sealed::GeneralPurpose16bitInstance;
-                Self::regs_gp16()
-                    .ccr(channel.raw())
-                    .modify(|w| w.set_ccr(value));
-            }
-
-            unsafe fn get_max_compare_value(&self) -> u16 {
-                use crate::timer::sealed::GeneralPurpose16bitInstance;
-                Self::regs_gp16().arr().read().arr()
-            }
-        }
+        impl_compare_capable_16bit!($inst);
```

Comment:
Replaced duplicated code here with an existing macro. Not sure why it wasn't used before? The code is identical.